xm-commons-domainevent-db/src/main/java/com/icthh/xm/commons/domainevent/db/domain/JpaEntityContext.java

```diff
@@ -0,0 +1,33 @@
+package com.icthh.xm.commons.domainevent.db.domain;
+
+import com.icthh.xm.commons.domainevent.domain.enums.DefaultDomainEventOperation;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.Serializable;
+import java.util.Map;
+
+@Data
+@EqualsAndHashCode
+@AllArgsConstructor
+@Builder
+public class JpaEntityContext {
+    private Object entity;
+    private Serializable id;
+    private Map<String, State> propertyNameToStates;
+    private DefaultDomainEventOperation domainEventOperation;
+
+    public String findPropertyStateValue(String property) {
+
+        if (propertyNameToStates.containsKey(property)) {
+            State state = propertyNameToStates.get(property);
+            return state.current() != null ? state.current().toString() : state.previous().toString();
```

Comment:
seems we will have NPE in case both current and previous values are null
consider using String.valueOf()?

Reply:
we cannot have both the current and the previous state null
if create entity -> previous = null
if delete entity -> current = null
if update entity ->  both states can not be null